clients/hadoopfs/src/main/java/io/lakefs/auth/AWSLakeFSTokenProvider.java

```diff
@@ -0,0 +1,151 @@
+package io.lakefs.auth;
+
+import com.amazonaws.Request;
+import com.amazonaws.auth.*;
+import io.lakefs.Constants;
+import io.lakefs.FSConfiguration;
+import io.lakefs.clients.sdk.ApiClient;
+import io.lakefs.clients.sdk.ApiException;
+import io.lakefs.clients.sdk.AuthApi;
+import io.lakefs.clients.sdk.model.AuthenticationToken;
+import org.apache.commons.codec.binary.Base64;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.URL;
+import java.util.HashMap;
+import java.util.Map;
+
+import com.amazonaws.util.json.JSONObject;
+import org.apache.hadoop.conf.Configuration;
+
+import io.lakefs.clients.sdk.model.LoginInformation;
+
+public class AWSLakeFSTokenProvider implements LakeFSTokenProvider {
+    STSGetCallerIdentityPresigner stsPresigner;
+    AWSCredentialsProvider awsProvider;
+    AuthenticationToken lakeFSAuthToken = null;
+    String stsEndpoint;
+    Map<String, String> stsAdditionalHeaders;
+    int stsExpirationInSeconds;
+    ApiClient lakeFSApi;
+
+    AWSLakeFSTokenProvider() {
```

Comment:
This is a very odd constructor.  It does nothing except change the scope.  Can we get rid of it?

Reply:
That's a default constructor I'm using it because the child class(es) `TemporaryAWSCredentialsLakeFSTokenProvider` initiates it and they also do some work before initiating it.